src/relations/openfga.py

```diff
@@ -114,24 +121,27 @@ def _on_openfga_store_created(self, event: OpenFGAStoreCreateEvent):
             logger.info(f"{event.relation.name} revoked, no store id")
             return
 
-        if event.token_secret_id:
-            secret = self.charm.model.get_secret(id=event.token_secret_id)
-            secret_content = secret.get_content()
-            token = secret_content["token"]
-        if event.token:
-            token = event.token
+        info = self.charm.openfga.get_store_info()
+        if not info:
+            logger.info(f"{event.relation.name} revoked, no store info found")
+            return
+
+        url_components = urlsplit(info.http_api_url)
+        scheme = url_components.scheme
+        address = url_components.hostname
+        http_port = url_components.port
 
         if self.charm.unit.is_leader():
```

Comment:
At the beginning of this method you check `if not self.charm.unit.is_leader()` and that evaluates to `True` you return. Do you then need to check again later `if self.charm.unit.is_leader()`? If this is not the case the unit already would have returned in the first evaluation

Reply:
Good point, thanks :)